app/models/dictionary.rb

```diff
@@ -491,18 +491,32 @@ def save_tags(tag_list)
   end
 
   def update_tags(tag_list)
-    current_tags = self.tags.pluck(:value)
-    tags_to_add = tag_list - current_tags
-    tags_to_remove = current_tags - tag_list
+    current_tags = self.tags.to_a
+    current_tag_values = current_tags.map(&:value)
+
+    tags_to_add = tag_list - current_tag_values
+    tags_to_remove = current_tag_values - tag_list
```

Comment:
```suggestion
    tags_to_remove = current_tag_values.reject { tag_list.include _1.value }
```

などにしたら、valueの配列でなくて、tagインスタンスの配列が取れそうです。

Reply:
ありがとうございます。
提案いただいたコードだと`undefined method `value' for "tag1":String`になってしまったので、修正したものを適用し、[current_tagsを使用して絞り込み](https://github.com/pubannotation/pubdictionaries/pull/71/commits/d805d6b5e931bf626601bfbbd09f1811239e7b8e)にて全体的にvalueではなくtagインスタンスを使ってロジック実行するようにしました。